server/src/main/java/io/crate/planner/optimizer/iterative/Memo.java

```diff
@@ -117,6 +120,19 @@ private LogicalPlan extract(LogicalPlan node) {
         return resolveGroupReferences(node, GroupReferenceResolver.from(this::resolve));
     }
 
+    public void addStats(int groupId, PlanStats stats) {
```

Comment:
`PlanStats` are stored for each group. If the stats for a specific group is updated, all depending groups are evicted.